File: problems/2248-intersection-of-multiple-arrays/solution.py
class Solution:
    def intersection(self, nums: List[List[int]]) -> List[int]:
        counts = {}
        for lst in nums:
            for num in lst:
                counts[num] = counts.get(num, 0) + 1
        result = []
        for num, count in counts.items():
            if count == len(nums):
                result.append(num)
        result.sort()
        return result

File: problems/2248-intersection-of-multiple-arrays/solution.cpp
class Solution {
public:
    vector<int> intersection(vector<vector<int>>& nums) {
        unordered_map<int, int> counts;
        for (const auto& vec : nums) {
            for (int num : vec) {
                counts[num]++;
            }
        }
        vector<int> result;
        int n = nums.size();
        for (const auto& pair : counts) {
            if (pair.second == n) {
                result.push_back(pair.first);
            }
        }
        sort(result.begin(), result.end());
        return result;
    }
};